common/odp/odp.go

```diff
@@ -84,3 +84,35 @@ func AddDatapathInterface(dpname string, ifname string) error {
 	_, err = dp.CreateVport(odp.NewNetdevVportSpec(ifname))
 	return err
 }
+
+func CreateVxlanVport(dpname string, name string, portno uint16) (odp.VportID, string, error) {
```

Comment:
It looks like `CreateDatapath` could be refactored to call this code, to reduce duplication.  Even the `ListenUDP` call is duplicated, although `CreateDatapath` leaves it open while `checkMTU` closes it before use.